### PR TITLE
[Doc] Add a note to build with "bazelisk" or "bazel==5.4.1".

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

When users building from source code with `bazel != 5.4.1`, users will encounter wired protobuf error:

```
 ERROR: /home/ubuntu/.cache/bazel/_bazel_ubuntu/2139a9ac91f0a6877cfd4920666e807a/external/com_google_protobuf/BUILD.bazel:82:6: Target '@@com_google_protobuf//src/google/protobuf:wrappers_proto' contains an error and its package is in error and referenced by '@@com_google_protobuf//:wrappers_proto'
    ERROR: Analysis of target '//:ray_pkg' failed; build aborted: Analysis failed
```

This is a big roadblock for developers and there is no troubleshooting documentation for it. This PR adds a note to recommend users always use `bazelisk` or a pinned version of `bazel==5.4.1`.

### Why `5.4.1` only?

The way bazel is used in ray is that in CI we symlink `bazel` to `bazelisk`, and the only version(5.4.1) that can be symlinked is hard-coded here: https://github.com/ray-project/ray/blob/master/.bazeliskrc

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]

File: doc/source/ray-contribute/development.rst [hunk 1]
```diff
@@ -143,6 +143,8 @@ To build Ray on Ubuntu, run the following commands:
   nvm install 14
   nvm use 14
 
+.. note::
+  We recommend using `bazelisk` for building. If you prefer to use `bazel`, ensure it is version `5.4.1`.
```
Comment (matthewdeng) [inline doc/source/ray-contribute/development.rst:147]: Can you explicitly call out that `bazelisk` is what `install-bazel.sh` installs?

Comment (matthewdeng) [inline doc/source/ray-contribute/development.rst:147]: Does it need to be exactly `5.4.1`? Can you explain a bit more about this (either in the note or in the PR description)? Not enough context right now to understand why it needs to be `5.4.1`.

Comment (aslonnie) [reply doc/source/ray-contribute/development.rst:147]: yes, and it is pinned her:
https://github.com/anyscale/product/blob/master/.bazelversion

different bazel version has different options / flags / skylark grammar support.

all versions should be pinned.

Comment (aslonnie) [reply doc/source/ray-contribute/development.rst:147]: the doc here probably should just say use `bazelisk`, or use the same version as `.bazelversion` specifies.

behavior of building with other bazel versions is undefined.

Comment (woshiyyya) [reply doc/source/ray-contribute/development.rst]: done

Comment (woshiyyya) [reply doc/source/ray-contribute/development.rst]: Updated the doc. Feel free to make some change if it's not accurate.

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**